src/embedding/kd.ts

```diff
@@ -332,6 +358,53 @@ export namespace EmbeddedKpiDashboard {
         return getEventType(obj) === GdcKdCommandType.SetSize;
     }
 
+    /**
+     * Data type of SetFilterContext command
+     */
+    export type SetFilterContextCommandData = IGdcKdMessageEnvelope<GdcKdCommandType.SetFilterContext, EmbeddedGdc.IFilterContextContent>;
+
+    /**
+     * Add or update the filter context
+     *
+     * Contract:
+     * - if filters is same with filters on the KD filter bar, then update the filters on the filter bar and apply the filters to dashboard
+     * - if filters is new one, then add it to the KD filter bar and apply to dashboard
+     * - in-case the AD can not apply the filters, a CommandFailed will be posted. The reason could be:
+     *   - exceed the limit number of filter items for selected filter values
```

Comment:
1. `KD`
2. `exceed the limit number of filter items for dashboard`

Reply:
```
For KD `CommandFailed` will be posted in cases:
filter is not existed
exceed the limit number of filter items
filter is existed but wrong elements input data
```